src/api/downloadFile.android.js

```diff
@@ -5,17 +5,21 @@ import type { Auth } from '../types';
 import { getAuthHeader } from '../utils/url';
 import userAgent from '../utils/userAgent';
 
-export default (url: string, auth: Auth) =>
+export default (url: string, auth: Auth, onSuccess?: () => void, onError?: () => void) =>
   RNFetchBlob.config({
     addAndroidDownloads: {
+      path: `${RNFetchBlob.fs.dirs.DownloadDir}/${url.split('/').pop()}`,
       useDownloadManager: true,
       // Optional, but recommended since android DownloadManager will fail when
       // the url does not contains a file extension, by default the mime type will be text/plain
       mime: 'text/plain',
       title: url.split('/').pop(),
     },
-  }).fetch('GET', url, {
-    'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8',
-    'User-Agent': userAgent,
-    Authorization: getAuthHeader(auth.email, auth.apiKey),
-  });
+  })
+    .fetch('GET', url, {
+      'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8',
+      'User-Agent': userAgent,
+      Authorization: getAuthHeader(auth.email, auth.apiKey),
+    })
+    .then(onSuccess)
+    .catch(onError);
```

Comment:
This is a good example where the `async`, `await` and `catch` paradigm works way better.
Just don't handle anything here and don't pass callbacks, leave the function as-is.

Reply:
Used `async`,` await` and `catch` 👍